src/main/java/Duke.java

```diff
@@ -1,10 +1,215 @@
+                System.out.println("Now you have " + listLength(tasks) + " tasks in the list.");
+                System.out.println("--------------------------------------------");
+                count1++;
+            }
+
+            else if (in.contains("event")) {
+                tasks[count1] = new Event(strings[0], strings[1]);
+                System.out.println("--------------------------------------------");
+                System.out.println("Got it. I've added this task: ");
+                System.out.println(tasks[count1].getPrintedLine());
+                System.out.println("Now you have " + listLength(tasks) + " tasks in the list.");
+                System.out.println("--------------------------------------------");
+                count1++;
+            }
+
+            else {
+                tasks[count1] = new Task();
+                tasks[count1].setDescription(in);
+                tasks[count1].setDone(false);
+                System.out.println("--------------------------------------------");
+                System.out.println("added: " + tasks[count1].getDescription());
+                System.out.println("--------------------------------------------");
+                count1++;
+            }
+            in = sc.nextLine();
+            strings = processInput(in);
+
+        } while (!in.equals("bye"));
+        System.out.println("--------------------------------------------");
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println("--------------------------------------------");
+    }
+}
+
+class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public Task(){};
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public void setDone(boolean done) {
+        isDone = done;
+    }
+
+    public void markAsDone() {
+        isDone = true;
+    }
+    public String getDescription() {
+        return this.description;
+    }
+    public String getStatusIcon() {
+        return (isDone ? "\u2713" : "\u2718"); //return tick or X symbols
     }
+
+    public String getPrintedLine(){
+        return ("[" + this.getStatusIcon() + "] " + this.description);
+    }
+
 }
+
+class Deadline extends Task {
+
+    protected String by;
+
+    public Deadline() {};
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    public String getPrintedLine() {
+        return "[D]" + super.getPrintedLine() + " (by: " + by + ")";
+    }
+}
+
+class ToDo extends Task {
+
+    public ToDo() {};
+    public ToDo(String description) {
+        super(description);
+    }
+
+    public String getPrintedLine() {
+        return "[E]" + super.getPrintedLine();
+    }
+}
+
+class Event extends Task {
```

Comment:
I like the classes you defined and their inherent.
maybe the code can be seperated into several class files.